sdk/eventhub/Azure.Messaging.EventHubs/src/Core/Guard.cs

```diff
@@ -64,6 +64,29 @@ internal static class Guard
             }
         }
 
+        /// <summary>
+        ///   Ensures that an argument's value is a string comprised of only whitespace, though
+        ///   <c>null</c> is considered a valid value.  An <see cref="ArgumentException" /> is thrown
+        ///   if that invariant is not met.
+        /// </summary>
+        ///
+        /// <param name="argumentName">The name of the argument being considered.</param>
+        /// <param name="argumentValue">The value of the argument to verify.</param>
+        ///
+        public static void ArgumentNotEmptyOrWhitespace(string argumentName,
+                                                       string argumentValue)
+        {
+            if (argumentValue == null)
```

Comment:
for null we are going to return instead of throwing. Was that intentional?

Reply:
Yes.   This one is "make sure that we're not empty or whitespace" for a few cases where we want to be sure that a legal null value was explicitly intended.   For example:  when creating an Event Producer and not requesting a partition.   I didn't want to assume that null/empty/whitespace was intended rather than developer error.